azurerm/internal/services/datashare/resource_arm_data_share_dataset_data_lake_gen1.go

```diff
@@ -0,0 +1,262 @@
+package datashare
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/datashare/mgmt/2019-11-01/datashare"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datashare/helper"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datashare/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datashare/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataShareDataSetDataLakeGen1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDataShareDataSetDataLakeGen1Create,
+		Read:   resourceArmDataShareDataSetDataLakeGen1Read,
+		Delete: resourceArmDataShareDataSetDataLakeGen1Delete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.DataShareDataSetID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DatashareDataSetName(),
+			},
+
+			"data_share_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DataShareID,
+			},
+
+			"data_lake_store": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: azure.ValidateDataLakeAccountName(),
+						},
+
+						"resource_group_name": azure.SchemaResourceGroupName(),
+
+						"subscription_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Sensitive:    true,
+							ValidateFunc: validation.IsUUID,
+						},
+					},
+				},
+			},
```

Comment:
is there any reason we can't have this as `data_lake_store_id` and grab the 3 properties from it? if not i think we should do that instead for better ux

Reply:
Hi kt, it for the `data_lake_store_id` could be subscription crossed , so we could not get the `data_lake_store_id` by its client, and we shall concatenate the id if we want only one field `data_lake_store_id`.

Reply:
Could you not get the subscription from the id? `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DataLakeStore/accounts/mydatalakeaccount`

Reply:
I think all the required bits on info are there?